src/decoding.rs

```diff
@@ -251,6 +252,29 @@ impl ProcessedBlockTrace {
                 .insert(val_k, updated_account_bytes.to_vec());
         }
 
+        // Remove any accounts that self-destructed.
```

Comment:
So I am a bit confused here. From the PR description 
> This PR adds a new field to the protocol to indicate that an account that existed before the current txn was destroyed in the current txn.

I thought the issue was that we were missing the account in the PartialTrie of the txn calling the SELFDESTRUCT, effectively causing the prover to hit a hash node when looking for it.
But this comment seems to imply we want to remove it? Or is it being applied after the actual SELFDESTRUCT call happened?

Reply:
Sorry I had missed this! That makes sense now, thanks!